application/frontend/controllers/MfaController.php

```diff
@@ -25,7 +25,10 @@ public function behaviors()
                 'rules' => [
                     [
                         'allow' => true,
-                        'roles' => ['@'],
+                        'matchCallback' => function () {
+                            $user = \Yii::$app->user->identity;
+                            return ( $user->isAuthScopeFull());
```

Comment:
This is a very minor issue, but you don't actually need the parentheses around this function call before returning the value. You can simply use...
```php
return $user->isAuthScopeFull();
```